packages/client/src/search/Search.ts

```diff
@@ -21,19 +22,29 @@ export class Search {
 
   async profiles(request: SearchProfilesQueryVariables): Promise<PaginatedResult<ProfileFragment>> {
     return buildPaginatedQueryResult(async (variables) => {
-      const result = await this.sdk.SearchProfiles(variables);
+      const response = await this.sdk.SearchProfiles(variables);
+      const result = response.data.result;
 
-      return result.data.result;
+      if (result.__typename === 'PublicationSearchResult') {
+        never('PublicationSearchResult are not expected in this query');
+      }
+
+      return result;
     }, request);
   }
 
   async publications(
     request: SearchPublicationsQueryVariables,
   ): Promise<PaginatedResult<CommentFragment | PostFragment>> {
     return buildPaginatedQueryResult(async (variables) => {
-      const result = await this.sdk.SearchPublications(variables);
+      const response = await this.sdk.SearchPublications(variables);
+      const result = response.data.result;
+
+      if (result.__typename === 'ProfileSearchResult') {
+        never('ProfileSearchResult are not expected in this query');
```

Comment:
what do you think about a solution like this?

Reply:
looks good, we can also go with `invariant(result.__typename === "PublicationSearchResult", ...)` to have just one line but that's nit